code/game/objects/machinery/computer/intel_computer.dm

```diff
@@ -51,10 +51,21 @@
 		STOP_PROCESSING(SSmachines, src)
 		printing = FALSE
 		printing_complete = TRUE
+		update_minimap_icon()
 		SSpoints.supply_points[faction] += supply_reward
 		SSpoints.dropship_points += dropship_reward
 		minor_announce("Classified transmission recieved from [get_area(src)]. Bonus delivered as [supply_reward] supply points and [dropship_reward] dropship points.", title = "TGMC Intel Division")
 		SSminimaps.remove_marker(src)
+		return
+	if (machine_stat & NOPOWER) /// checks for lack of power and shuts down
```

Comment:
the way you've set this up, you can still finish the intel computer without power on the last cycle, since it adds (and checks) progress above.

Reply:
done & made an else as per the second request